packages/compiler/lib/decorators.ts

```diff
@@ -888,3 +888,71 @@ export function getOverloadedOperation(
 ): OperationType | undefined {
   return program.stateMap(overloadsOperationKey).get(operation);
 }
+
+const projectedNameKey = Symbol("overloadedByKey");
+const projectedNameDecorator = createDecoratorDefinition({
+  name: "@projectedName",
+  target: "Any",
+  args: [{ kind: "String" }, { kind: "String" }],
+} as const);
+
+/**
+ * `@projectedName` - Indicate that this entity should be renamed according to the given projection.
+ * @param context DecoratorContext
+ * @param target The that should have a different name.
+ * @param projectionName Name of the projection (e.g. "toJson", "toCSharp")
+ * @param projectedName Name of the type should have in the scope of the projection specified.
+ */
+export function $projectedName(
+  context: DecoratorContext,
+  target: Type,
+  projectionName: string,
```

Comment:
Do we want to start with a closed set of projection names?

Reply:
hhm maybe, just to make sure we set the right guideline to get started